### DISCUSSION TITLE
Stop child route render after error in parent loader

### DISCUSSION THREAD

**Discussion by hosmelq** [TanStack/router#653] [Q&A]
Hi, is there a method to halt the rendering of a child route in case of an error in the parent route's loader?

For example. I am using react-query to ensure data is valid, I have a default error component that renders if the status code is 404. That part works, but the child route re-renders.

```ts
loader({context, params}) {
  return context.queryClient.ensureQueryData({
    queryKey: ['organization', params.organizationId],
    queryFn: () => loadOrganization(params.organizationId),
  })
}
```

**Comment by tannerlinsley** [comment]
This should already be happening. Can you provide a reproduction in code sandbox or blitz?

**Reply by hosmelq** [reply]
I apologize for my misunderstanding. To clarify, I am referring to the loader of the child router that still runs even if the parent loader fails and the error component is shown.

Here is a reproduction.

https://codesandbox.io/p/sandbox/zealous-yalow-4kknrn

**Reply by pawelblaszczyk5** [reply]
Afaik, they both start running at the same time after router internals match them. They should be independent of each other

**Comment by tannerlinsley** [comment]
They do need to run in parallel by default, so it sounds like you want a way to have your child loader wait for the parent in a serial pattern. Correct?

**Reply by hosmelq** [reply]
Yes. For example, this path `/organization/1/posts` loads the organization data and the posts for that organization, but if the organization does not exist, I don't want to make the posts request since it will return not fount too.

**Reply by tannerlinsley** [reply]
I can’t remember if you can access the parent promise already, but if you can, you would just await it and then fetch your child data. If you can’t access it, then I’ll make it so you can.

**Reply by tannerlinsley** [reply]
Double checked and that promise isn't there, but it will be in the next release under the `parentMatchPromise` property on the object that gets passed to the loader function.

**Reply by visualjerk** [reply]
I am unsure about how to use `parentMatchPromise`. Can I expect it to be defined in this scenario?

- `parent.tsx`

```ts
import { Outlet, createFileRoute } from '@tanstack/react-router';

export const Route = createFileRoute('/parent')({
  async loader() {
    // ... loading data
    return { data }
  },
  ...
});
```

- `parent.child.tsx`

```ts
import { createFileRoute } from '@tanstack/react-router';

export const Route = createFileRoute('/parent/child')({
  async loader({ parentMatchPromise }) {
    await parentMatchPromise; // <-- undefined
  },
  ...
});
```

**Reply by schiller-manuel** [reply]
it should be defined, yes. can you provide a reproducer where it is `undefined`?

**Reply by visualjerk** [reply]
Sure, here is a minimal example: https://stackblitz.com/edit/tanstack-router-rekhkq?file=src%2Froutes%2Fparent.child.tsx

In order to reproduce the issue, try to navigate to "/parent/child".

I can also create a new issue, if that is unintended behavior.

**Reply by schiller-manuel** [reply]
yes please create a new issue

**Reply by ibrahim-ytl** [reply]
Is this issue fixed? I am still encountering always undefined for `parentMatchPromise`. I am using version `^1.45.10`

**Reply by schiller-manuel** [reply]
@ibrahim-ytl please upgrade to the latest version. if the issue still persists, please create a new Github issue including a complete minimal example.
